prom_wrapper.coffee

```diff
@@ -0,0 +1,114 @@
+prom = require('prom-client')
+registry = require('prom-client').register
+metrics = {}
+
+
+optsKey = (opts) ->
+	keys = Object.keys(opts)
+	return '' if keys.length == 0
+
+	keys = keys.sort()
+
+	hash = '';
+	for key in keys
+		hash += "," if hash.length
+		hash += "#{key}:#{opts[key]}"
+
+	return hash
+
+extendOpts = (opts, labelNames) ->
+	for label in labelNames
+		opts[label] ||= ''
+	return opts
+
+optsAsArgs = (opts, labelNames) ->
+	args = []
+	for label in labelNames
+		args.push(opts[label] || '')
+	return args
+
+
+PromWrapper =
+	ttlInMinutes: 0
+	registry: registry
+
+	metric: (type, name) ->
+		registry.getSingleMetric(name) || new MetricWrapper(type, name)
+
+	collectDefaultMetrics: prom.collectDefaultMetrics
+
+
+class MetricWrapper
+	constructor: (type, name) ->
+		metrics[name] = this
+		@name = name
+		@instances = {}
```

Comment:
I know that in practice this is unlikely to be a problem in practice but I will just flag up that  the good practice of using  a Map for `metrics[]` and `instances[]` to avoid problems if there is a metric using a reserved javascript name (I am guilty of using  a hash myself many times).